daemon/src/setup_taker.rs

```diff
@@ -0,0 +1,213 @@
+use crate::{
+    db,
+    model::{
+        cfd::{Cfd, CfdState, Dlc, Order, OrderId, Role},
+        Usd,
+    },
+    oracle, setup_contract,
+    tokio_ext::spawn_fallible,
+    wallet,
+    wire::{self, SetupMsg},
+};
+use anyhow::Context;
+use anyhow::Result;
+use async_trait::async_trait;
+use futures::{
+    channel::mpsc::{self, UnboundedSender},
+    future, SinkExt,
+};
+use maia::secp256k1_zkp::schnorrsig;
+use xtra::prelude::MessageChannel;
+use xtra_productivity::xtra_productivity;
+
+pub struct Actor<W> {
+    cfd: (Order, Usd),
+    n_payouts: usize,
+    db: sqlx::SqlitePool,
+    wallet: xtra::Address<W>,
+    maker: Box<dyn MessageChannel<wire::TakerToMaker>>,
+    get_announcement: Box<dyn MessageChannel<oracle::GetAnnouncement>>,
+    monitor_attestation: Box<dyn MessageChannel<oracle::MonitorAttestation>>,
+    on_completed: Box<dyn MessageChannel<Completed>>,
+    oracle_pk: schnorrsig::PublicKey,
+    setup_msg_sender: Option<UnboundedSender<SetupMsg>>,
+}
+
+impl<W> Actor<W> {
+    #[allow(clippy::too_many_arguments)]
+    pub fn new(
+        cfd: (Order, Usd),
+        n_payouts: usize,
+        db: sqlx::SqlitePool,
+        wallet: xtra::Address<W>,
+        maker: Box<dyn MessageChannel<wire::TakerToMaker>>,
+        get_announcement: impl MessageChannel<oracle::GetAnnouncement> + 'static,
+        monitor_attestation: impl MessageChannel<oracle::MonitorAttestation> + 'static,
+        on_completed: impl MessageChannel<Completed> + 'static,
+        oracle_pk: schnorrsig::PublicKey,
+    ) -> Self {
+        Self {
+            cfd,
+            n_payouts,
+            db,
+            wallet,
+            maker,
+            get_announcement: Box::new(get_announcement),
+            monitor_attestation: Box::new(monitor_attestation),
+            on_completed: Box::new(on_completed),
+            oracle_pk,
+            setup_msg_sender: None,
+        }
+    }
+}
+
+#[xtra_productivity]
+impl<W> Actor<W>
+where
+    W: xtra::Handler<wallet::Sign> + xtra::Handler<wallet::BuildPartyParams>,
+{
+    fn handle(&mut self, _: Accepted, ctx: &mut xtra::Context<Self>) -> Result<()> {
+        let (order, quantity) = &self.cfd;
+        let order_id = order.id;
+
+        tracing::info!(%order_id, "Order got accepted");
+
+        let mut conn = self.db.acquire().await?;
+        let cfd = Cfd::new(order.clone(), *quantity, CfdState::accepted());
+        db::append_cfd_state(&cfd, &mut conn).await?;
+        // TODO:
+        // Updating the cfd feed is missing i.e.
+        //   `update_sender.send(db::load_all_cfds(conn).await?)?;`
+        // I see two options:
+        // - make `update_sender` a mpmc (via `crossbeam` for example).
+        // - send a message to the `taker_cfd` to update the feed.
```

Comment:
:question: What do you think about this problem, @thomaseizinger?

Reply:
Can we move the update back into `taker_cfd::Actor` and you just send it a message back? Will require another `MessageChannel` but for now, that is probably the least invasive thing.